btree/btree.c

```diff
@@ -2621,21 +2637,27 @@ static int64_t m0_btree_put_root_split_handle(struct m0_btree_op *bop,
 	/* if we need to update vec_count for root slot, update at this place */
 
 	node_done(&node_slot, bop->bo_tx, true);
-	node_fix(curr_level->l_node, bop->bo_tx);
+	node_fix(lev->l_node, bop->bo_tx);
 
 	/* Increase height by one */
 	tree->t_height++;
 
+
+	node_put(lev->l_alloc);
+	lev->l_alloc = NULL;
```

Comment:
Do we need to set l_alloc to NULL here?

Reply:
Yes , else it will get freed in level_cleanup